web/impact/impact/v1/views/reserve_office_hour_view.py

```diff
@@ -213,8 +213,11 @@ def _succeed(self):
         self.timecard_info = {
             "finalist_first_name": self.target_user.first_name,
             "finalist_last_name": self.target_user.last_name,
+            "finalist_email": self.target_user.email,
             "topics": self.message,
-            "startup": startup_name}
+            "startup": startup_name,
+            "calendar_data": self.calendar_data,
```

Comment:
This is unsafe. 
Since it `self.calendar_data` is set in `prepare_email_notification`, this will break if the order in which these functions run is ever changed. If we want to make this change, let's do it a little more cautiously. 


I would suggest that `self.get_calendar_data` should check to see if `hasattr(self, "_calendar_data)`. If so, return it. If not, compute the calendar data and set `self._calendar_data=calendar_data`. 

Having done that, you can safely call `self.get_calendar_data()` on lines 219 and 191 (and you don't need the assignment on line 186)